dashboard/app/controllers/aichat_controller.rb

```diff
@@ -14,49 +16,52 @@ def chat_completion
       return render status: :bad_request, json: {}
     end
 
+    response_body = get_response_body
+    response_body[:session_id] = log_chat_session(response_body[:messages])
+    render(status: :ok, json: response_body)
+  end
+
+  private def get_response_body
     # Check for profanity
     locale = params[:locale] || "en"
-    filter_result = ShareFiltering.find_failure(params[:newMessage], locale)
+    filter_result = ShareFiltering.find_failure(params[:newMessage][:chatMessageText], locale)
     if filter_result&.type == ShareFiltering::FailureType::PROFANITY
-      new_messages = [
-        {
-          role: 'user',
-          content: params[:newMessage],
-          status: SharedConstants::AI_INTERACTION_STATUS[:PROFANITY_VIOLATION]
-        }
+      messages = [
+        get_user_message(SharedConstants::AI_INTERACTION_STATUS[:PROFANITY_VIOLATION])
       ]
-      session_id = log_chat_session(new_messages)
-      return render(
-        status: :ok,
-        json: {
-          status: SharedConstants::AICHAT_ERROR_TYPE[:PROFANITY_USER],
-          flagged_content: filter_result.content,
-          session_id: session_id
-        }
-      )
+
+      return {
+        messages: messages,
+        flagged_content: filter_result.content,
+      }
+    end
+
+    messages_for_model = params.to_unsafe_h[:storedMessages].filter do |message|
+      message[:status] == SharedConstants::AI_INTERACTION_STATUS[:OK] &&
```

Comment:
I think we already filter for messages in `conversationMessages` on the front-end for user and assistant responses. Is this for redundancy?

Reply:
haha yeah, good catch -- it felt safer to me to be redundant in the absence of any testing of our front end API currently, ok with you?

Reply:
Sure - just confirming.